sklearn/model_selection/_validation.py

```diff
@@ -1235,6 +1235,21 @@ def _check_is_permutation(indices, n_samples):
     return True
 
 
+@validate_params(
+    {
+        "estimator": [HasMethods("fit")],
+        "X": ["array-like"],
```

Comment:
Sparse matrices are also valid. Please also update the descr of ``X`` in the docstring.
```suggestion
        "X": ["array-like", "sparse matrix"],
```